### PR TITLE
Make the CPack DEB packet and the PPA build equal using 

### PR DESCRIPTION
**Pull request by daschuer**
This PR replaces https://github.com/mixxxdj/mixxx/pull/2790 by a single file solution, that compares to the CPack PPA upload solution. 
Finally the PPA And the CPack deb files have the same content. Also the debian control file contains the same data but in a different order. 

~~This PR is on top of https://github.com/mixxxdj/mixxx/pull/3570 which should be merged first.~~ Done

### CODE REVIEW

**Comment by Be-ing** [comment]
Please rebase on 2.3 now that #3570 has been merged.

File: packaging/CPackDebInstall.cmake [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+find_program( CPACK_DEBIAN_DEBHELPER dh_prep )
+if( NOT CPACK_DEBIAN_DEBHELPER )
+    message( FATAL_ERROR "debhelper not found, required for cpack -G DEB" )
+endif()
+
+find_program( CPACK_DEBIAN_MARKDOWN markdown )
+if( NOT CPACK_DEBIAN_MARKDOWN )
+   message( FATAL_ERROR "markdown not found, required for cpack -G External -D DEB_UPLOAD_PPA=true" )
+endif()
```
Comment (Be-ing) [inline packaging/CPackDebInstall.cmake:13]: Is this required for `cpack -G DEB`?

File: packaging/CPackDebInstall.cmake [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+configure_file(${CPACK_TOPLEVEL_DIRECTORY}/${CPACK_PACKAGE_FILE_NAME}/debian/control.in
+               ${CPACK_TOPLEVEL_DIRECTORY}/${CPACK_PACKAGE_FILE_NAME}/debian/control
+               @ONLY)
+file(REMOVE ${CPACK_TOPLEVEL_DIRECTORY}/${CPACK_PACKAGE_FILE_NAME}/debian/control.in)
+
+file(COPY ${CPACK_DEBIAN_SOURCE_DIR}/res/linux/mixxx-usb-uaccess.rules
+    DESTINATION ${CPACK_TOPLEVEL_DIRECTORY}/${CPACK_PACKAGE_FILE_NAME}/debian)
+file(RENAME
+    ${CPACK_TOPLEVEL_DIRECTORY}/${CPACK_PACKAGE_FILE_NAME}/debian/mixxx-usb-uaccess.rules
+    ${CPACK_TOPLEVEL_DIRECTORY}/${CPACK_PACKAGE_FILE_NAME}/debian/mixxx.mixxx-usb-uaccess.udev)
```
Comment (Be-ing) [inline packaging/CPackDebInstall.cmake:32]: Can these be combined to one `file(COPY)` command?

**Comment by Be-ing** [comment]
Just a few minor comments about the CMake scripts. I don't know much about Debian packaging but I trust you that this works.

**Comment by Be-ing** [comment]
This doesn't need to be done in this PR, but could you make DEB builds available as GitHub Actions artifacts?

**Comment by daschuer** [comment]
We have them already.

File: packaging/CPackDebInstall.cmake (hunk 1 above)
Comment (daschuer) [reply packaging/CPackDebInstall.cmake:13]: Yes, as well as for DEB_UPLOAD

File: packaging/CPackDebInstall.cmake (hunk 2 above)
Comment (daschuer) [reply packaging/CPackDebInstall.cmake:32]: No, a file can only copied to a directory.

**Comment by daschuer** [comment]
Done

**Comment by daschuer** [comment]
All green finally.

**Review by uklotzde — APPROVED** [review]
@Be-ing Ready to merge?

**Pull request merged**